src/Classes/TradeQueryGenerator.lua

```diff
@@ -402,10 +402,19 @@ function TradeQueryGeneratorClass:GenerateModWeights(modsToTest)
 				logToFile("Failed to test %s mod: %s", self.calcContext.itemCategory, modLine)
 			end
 
+			local baseOutput = self.calcContext.baseOutput
 			local output = self.calcContext.calcFunc({ repSlotName = self.calcContext.slot.slotName, repItem = self.calcContext.testItem }, {})
-			local meanDPSDiff = (GlobalCache.useFullDPS and output.FullDPS or m_max(output.TotalDPS, m_max(output.TotalDotDPS,output.CombinedDPS)) or 0) - (self.calcContext.baseDPS or 0)
-			if meanDPSDiff > 0.01 then
-				table.insert(self.modWeights, { tradeModId = entry.tradeMod.id, weight = meanDPSDiff / modValue, meanDPSDiff = meanDPSDiff, invert = entry.sign == "-" and true or false })
+			local meanStatDiff = 0
+			for _, statTable in ipairs(self.calcContext.options.statWeights) do
+				if statTable.stat == "FullDPS" and not GlobalCache.useFullDPS then
+					meanStatDiff = meanStatDiff + m_max((output.TotalDPS or 0) / (baseOutput.TotalDPS or 0), m_max((output.TotalDotDPS or 0) / (baseOutput.TotalDotDPS or 0), (output.CombinedDPS or 0) / (baseOutput.CombinedDPS or 0))) * statTable.weightMult
+				else
+					meanStatDiff = meanStatDiff + ( output[statTable.stat] or 0 ) / ( baseOutput[statTable.stat] or 0 ) * statTable.weightMult
+				end
+			end
+			meanStatDiff = meanStatDiff * 1000 - (self.calcContext.baseStatValue or 0)
```

Comment:
This causes some problems when any of the stats taken from baseOutput is 0 - blank (nan) weights in searches or failing to generate searches altogether. Also taking the max of the ratios for full DPS mode leads to disproportionately high weights on very insignificant stats (i.e. increased fire damage on a cold based crit build with a tiny bit of added fire damage, because the relative increase on the tiny ignite is large).

I'm working on top of this branch atm and my current solution is:
```
function TradeQueryGeneratorClass.WeightedRatioOutputs(baseOutput, newOutput, statWeights)
	local meanStatDiff = 0
	local function ratioModSums(...)
		local baseModSum = 0
		local newModSum = 0
		for _, mod in ipairs({ ... }) do
			baseModSum = baseModSum + baseOutput[mod] or 0
			newModSum = newModSum + newOutput[mod] or 0
		end
		if baseModSum == 0 then
			return newModSum
		else
			return newModSum / baseModSum
		end
	end
	for _, statTable in ipairs(statWeights) do
		if statTable.stat == "FullDPS" and not GlobalCache.useFullDPS then
			meanStatDiff = meanStatDiff + ratioModSums("TotalDPS", "TotalDotDPS", "CombinedDPS") * statTable.weightMult
		else
			meanStatDiff = meanStatDiff + ratioModSums(statTable.stat) * statTable.weightMult
		end
	end
	return meanStatDiff
end
```
and then
```suggestion
	local meanStatDiff = TradeQueryGeneratorClass.WeightedRatioOutputs(baseOutput, output, self.calcContext.options.statWeights) * 1000 - (self.calcContext.baseStatValue or 0)
```
(and equivalently in a few other places were this logic is duplicated)